examples/matching-engine/src/contract.rs

```diff
@@ -280,6 +318,10 @@ impl MatchingEngine {
         Ok(())
     }
 
+    /// For a specific level of price, look at all the orders and find the one that
+    /// matches or emit an error.
+    /// When that order is found, then the cancellation is applied to it.
+    /// Then the information is emitted for the handling of this operation.
```

Comment:
Doesn't it find the order by ID, rather than price level?

I'd either not mention the error, or explain the error conditions.

(Also: 3rd person in method comments.)